modules/admin_manual/pages/configuration/user/oidc/index.adoc

```diff
@@ -3,56 +3,215 @@
 :toclevel: 2
 :openid-connect-frontchannel-logout-url: https://openid.net/specs/openid-connect-frontchannel-1_0.html
 :openid-connect-url: https://openid.net/connect/
+:openid-app-marketplace-url: https://marketplace.owncloud.com/apps/openidconnect
+:openid-app-github-url: https://github.com/owncloud/openidconnect
+:examplesdir: ../../../../examples/
 
 == Introduction
 
 "OpenID Connect 1.0 is a simple identity layer on top of the OAuth 2.0 protocol. It allows Clients to verify the identity of the End-User based on the authentication performed by an Authorization Server, as well as to obtain basic profile information about the End-User in an interoperable and REST-like manner.
 OpenID Connect allows clients of all types, including Web-based, mobile, and JavaScript clients, to request and receive information about authenticated sessions and end-users. The specification suite is extensible, allowing participants to use optional features such as encryption of identity data, discovery of OpenID Providers, and session management, when it makes sense for them."
 -- {openid-connect-url}, "What is OpenID Connect?"
 
+//ownCloud-specific OIDC description and benefits
+== OIDC Benefits
+
+== OIDC App Marketplace Link
+
+// Not currently available
+* https://marketplace.owncloud.com/apps/openidconnect
+
 == Prerequisites
 
-A distributed Memcache setup is required to operate this app - such as Redis or Memcached properly.
-Please follow xref:configuration/server/caching_configuration.adoc[the documentation] on how to set up caching.
+The OpenID Connect app has several mandatory requirements, these are:
+
+* A minimum PHP version of 7.0, with the CURL and JSON extensions.
+* xref:configuration/server/caching_configuration.adoc[A distributed memory cache setup], such as _Redis_ or _Memcached_.
+This is because information about tokens need to be cached to prevent roundtrips to the identity provider. 
+Additionally, in clustered setups this information needs to be available for all nodes, which necessitates a distributed cache.
++
+NOTE: If a memory cache is not available, then you will be prevented from using the ownCloud UI, and see the following message: "_A real distributed mem cache setup is required_" in the ownCloud UI.
+
+== Recommended Options & Extensions
+
```

Comment:
@settermjd following info:
- Please see https://doc.owncloud.com/server/10.4/admin_manual/installation/server_prep_ubuntu_18.04.html which covers a lot of the table you have described plus some additional infos. You may link to it
- The GMP library is already part of the base installation as described in the link above. We could add a info to the link above regarding "what are the differences between the BCMath and GMP libraries", see https://stackoverflow.com/questions/6961685/arbitrary-precision-math-in-php
- You can see which php libraries are enabled with a command described in the link above
- Please see the notes in the link above regarding libsodium
- mcrypt I: the mcrypt library is not necessary (outdated). I have added a note in the link above if someone really needs it. I will do a improvement of the content to make it more clear.
https://github.com/phpseclib/phpseclib/issues/1028#issuecomment-242739527 ... "mcrypt is only used if it's available. If mcrypt is not available either a pure-PHP implementation is used or OpenSSL is used. The prioritization is as follows: OpenSSL > mcrypt > pure-PHP. mcrypt and OpenSSL are loads faster than the pure-PHP implementation". And "mcrypt offers a 45x speedup over the internal mode. OpenSSL offers a 6.5x speedup over mcrypt"
- mcrypt II: "php7.x.x has openssl compiled within the php core and doesn't need an external extension to enable it" AND "there is some problem with openssl provided with php 7.1.x, but the problem is resolved at for php versions after 7.2.x as they have inbuilt openssl 1.1.1" which is another good reason for using php7.3 (7.2 is already eos!). See https://stackoverflow.com/questions/50335975/there-is-no-openssl-so-extension-with-php7-2-x

Reply:
I have created a PR to improve the `server_prep_ubuntu_18.04` documentation